### PR TITLE
Some minor fixes and optimizations

### PR DESCRIPTION
**Pull request by Sirbucket**
Nothing huge just a sweep of anonymous functions and a few rogue bugs in /functions

### CODE REVIEW

File: functions/api2.lua [hunk 1]
```diff
@@ -427,20 +427,21 @@ tinsert(Details.API_Description.namespaces[1].api, {
...
 		return phases
-	end	
-	
+	end
+
 	local phaseData = combatObject.PhaseData
-	
+
 	for phaseChangeId, phaseTable in ipairs(phaseData) do
+		local phaseTable = phaseData [phaseChangeId]
```
Comment (Flamanis) [inline functions/api2.lua:440]: This is useless. Ipairs already returns the value as a local

File: functions/api2.lua [hunk 2]
```diff
@@ -927,22 +928,22 @@ function Details.UnitDamageSpellInfo (unitId, spellId, isLiteral, segment)
...
 		return spellInfo
 	end
-	
+
 	local spellObject, spellId = getSpellObject (playerObject, spellId, isLiteral)
 	if (not spellObject) then
 		return spellInfo
 	end
 
+	local spellName = GetSpellInfo(spellId)
```
Comment (Flamanis) [inline functions/api2.lua:942]: Moving this out of the below check is unneded

File: functions/api2.lua [hunk 3]
```diff
@@ -1658,22 +1659,22 @@ function Details.UnitHealingSpellInfo (unitId, spellId, isLiteral, segment)
...
```
Comment (Flamanis) [inline functions/api2.lua:1673]: Same as before

File: functions/autorun.lua [hunk 4]
```diff
@@ -54,14 +54,15 @@ autoRunCodeEventFrame.OnEventFunc = function(self, event)
     if (event == "PLAYER_SPECIALIZATION_CHANGED") then
         --create a trigger for the event, many times it is triggered more than once
         --so if the event is triggered a second time, it will be ignored
-        local newTimer = C_Timer.NewTimer(1, function()
-            Details:DispatchAutoRunCode("on_specchanged")
-            
+		local spec_changed_func = function()
```
Comment (Flamanis) [inline functions/autorun.lua:57]: These changes do little to change the functionality of these functions. They reduce readability imo.

**Review by Flamanis — COMMENTED** [review]
The vast majority of changes are either removing anonymous functions, which could potentially save microseconds once every startup or infrequent operation, if that. I would bet it would be at the cost of memory overall. And automatic whitespace removal. The only change i see that would be any improvement would be the consolidation of the GetFramework calls, but even then, thats such an i frequent operation that it is not an issue.

File: functions/api2.lua (hunk 2 above)
Comment (Flamanis) [reply functions/api2.lua:942]: I actually do take this back, it would remove a nil for spell name in a lower check. Minor fix but a fix nonetheless

**Comment by Sirbucket** [comment]
alr

**Comment by Flamanis** [comment]
I don't want this to disaude you. You did find an actual issue. The SpellName ones are absolutely correct to change.

**Pull request closed without merging**